src/client/LanguageClient.ts

```diff
@@ -84,19 +85,50 @@ async function addSubscriptions(context: ExtensionContext) {
     }),
   );
 
+  function isValidDirectory(filePath: string): boolean {
+    try {
+      const stats = fs.statSync(filePath);
+      return stats.isDirectory();
+    } catch (error) {
+      return false;
+    }
+  }
+
   // Deploy command from user
   context.subscriptions.push(
     commands.registerCommand('metaed.deploy', () => {
       (async () => {
+        const deployDirectoryPath = getOdsApiDeploymentDirectory();
+        const implementationFolderPath = path.join(deployDirectoryPath, 'Ed-Fi-ODS-Implementation');
+        const odsFolderPath = path.join(deployDirectoryPath, 'Ed-Fi-ODS');
+        const drivePattern = /[a-zA-Z]:/;
         if (!acceptedLicense()) {
           await showErrorNotification(
             'You must first accept the Ed-Fi License Agreement under File -> Preferences -> Settings.',
           );
           return;
         }
 
-        if (getOdsApiDeploymentDirectory() === '') {
-          await showInfoNotification('To deploy, set Ods Api Deployment Directory under File -> Preferences -> Settings.');
+        if (deployDirectoryPath === '') {
+          await showErrorNotification('To deploy, set Ods Api Deployment Directory under File -> Preferences -> Settings.');
+          return;
+        }
+        if (!isValidDirectory(deployDirectoryPath)) {
+          await showErrorNotification(
+            'Directory path not found, set proper Ods Api Deployment Directory under File -> Preferences -> Settings.',
+          );
+          return;
+        }
+        if (drivePattern.test(deployDirectoryPath) && !deployDirectoryPath.substring(0, 3).endsWith(path.sep)) {
```

Comment:
I think you can remove this. It's covered by last check, and isn't cross-platform. The original ticket had an issue with "C:" versus "C:\", but while strange is actually expected behavior. [Omitting the back slash means current directory on that drive](https://learn.microsoft.com/en-us/windows/win32/fileio/naming-a-file#fully-qualified-vs-relative-paths)

Reply:
@bradbanister 
If I remove that check the following use case is failing : Have "EdFi-ODS and Implementation" folders on C drive directly. On extension setting just give "C:" for Deploy directory, the deployment is running. But not creating any folders under "EdFi-ODS"(no artifacts getting created).